src/oops/SOLID/singleResponsibilityPrinciple/before/Employee.java

```diff
@@ -66,21 +66,7 @@ public String getFullName(){
     public  void save(){
         try {
         	Employee employee =this;
-            StringBuilder sb = new StringBuilder();
-            sb.append("### EMPLOYEE RECORD ####");
-            sb.append(System.lineSeparator());
-            sb.append("NAME: ");
-            sb.append(employee.firstName + " " + employee.lastName);
-            sb.append(System.lineSeparator());
-            sb.append("POSITION: ");
-            sb.append(employee.getClass().getTypeName());
-            sb.append(System.lineSeparator());
-            sb.append("EMAIL: ");
-            sb.append(employee.getEmail());
-            sb.append(System.lineSeparator());
-            sb.append("MONTHLY WAGE: ");
-            sb.append(employee.monthlyIncome);
-            sb.append(System.lineSeparator());
+            StringBuilder sb = serializeEmployee(employee);
 
             Path path = Paths.get(employee.getFullName()
```

Comment:
Let's say you want to connect to Mysql DB and want to commit changes. So first you will need the cursor and then you will do the commit. So here save method is doing the connection first and then it is going to do the commit. You can make the connection in a private method and commit.